nautobot/extras/models/change_logging.py

```diff
@@ -28,10 +28,36 @@ class ChangeLoggedModel(models.Model):
 
     created = models.DateTimeField(auto_now_add=True, blank=True, null=True)
     last_updated = models.DateTimeField(auto_now=True, blank=True, null=True)
+    created_by = models.ForeignKey(
+        to=settings.AUTH_USER_MODEL,
+        on_delete=models.SET_NULL,
+        blank=True,
+        null=True,
+        related_name="+",
+    )
+    last_updated_by = models.ForeignKey(
+        to=settings.AUTH_USER_MODEL,
+        on_delete=models.SET_NULL,
+        blank=True,
+        null=True,
+        related_name="+",
+    )
 
     class Meta:
         abstract = True
 
+    def save(self, *args, **kwargs):
+        from nautobot.extras.choices import ObjectChangeActionChoices
+
+        # call save() first so that the ObjectChange record exists
+        super().save(*args, **kwargs)
+
+        most_recent_change = get_changes_for_model(self).first()
+        if most_recent_change and most_recent_change.action == ObjectChangeActionChoices.ACTION_CREATE:
+            self.created_by = most_recent_change.user
+        self.last_updated_by = most_recent_change.user
```

Comment:
Would it make any sense to instead do this as part of the change_logging decorator, instead of the object save()? Just thinking about bulk operations and avoiding the need to repeatedly look up the ObjectChange and associated User for each and every object created/modified.